server/lib/changelly/changelly.ctrl.js

```diff
@@ -59,10 +59,34 @@ ChangellyCtrl.getExchangeAmount = (req, res) => {
   return ChangellyCtrl.request('getExchangeAmount', req.params, (err, data) => {
     if (err) {
       console.log('Error: ', err)
+      res.send(err)
     } else {
       res.send(data)
     }
   })
 }
 
+ChangellyCtrl.generateAddress = (req, res) => {
+  return ChangellyCtrl.request('generateAddress', req.params, (err, data) => {
+    if (err) {
+      console.log('Error: ', err)
+      res.send(err)
+    } else {
+      res.send(data)
+    }
+  })
+}
+
+ChangellyCtrl.internal.generateAddress = (params) => {
+  return new Promise((fulfill, reject) => {
+    ChangellyCtrl.request('generateAddress', params, (err, data) => {
+      if (data.err) {
+        console.log('Error: ', data.err)
+        reject(new Error(data.err))
```

Comment:
you may need a return after this line, otherwise it will call both reject and fulfill when there is an error.